repos/system_upgrade/common/actors/checkmountoptions/libraries/checkmountoptions.py

```diff
@@ -0,0 +1,75 @@
+from leapp.models import StorageInfo
+
+
+def inhibit_upgrade_due_var_with_noexec(mountpoint, found_in_fstab=False):
+    summary = (
+        'Leapp detected that the {0} mountpoint is mounted with the "noexec" option, '
+        'which prevents binaries necessary for the upgrade from being executed. '
+        'The upgrade process cannot continue with {0} mounted using the "noexec" option.'
+    )
+
+    if found_in_fstab:
+        hint = (
+            'Temporarily remove the "noexec" option from {0} entry in /etc/fstab until the system is upgraded, '
+            'and remount the partition without the "noexec" option.'
+        )
+        related_resource = [reporting.RelatedResource('file', '/etc/fstab')]
+    else:
+        hint = (
+            'Remount {0} without the noexec option and make sure the change is persistent'
+            'during the entire in-place upgrade process.'
+        )
+        related_resource = []
+
+    reporting.create_report([
+        reporting.Title(
+            'Detected partitions mounted with the "noexec" option, preventing a successful in-place upgrade.'
+        ),
+        reporting.Summary(summary.format(mountpoint)),
+        reporting.Remediation(hint=hint.format(mountpoint)),
+        reporting.RelatedResource('file', '/etc/fstab'),
+        reporting.Severity(reporting.Severity.HIGH),
+        reporting.Tags([reporting.Tags.FILESYSTEM]),
+        reporting.Flags([reporting.Flags.INHIBITOR]),
+    ] + related_resource)
+
+
+def find_mount_entry_with_mountpoint(mount_entries, mountpoint):
+    for mount_entry in mount_entries:
+        if mount_entry.mount == mountpoint:
+            return mount_entry
+    return None
+
+
+def find_fstab_entry_with_mountpoint(fstab_entries, mountpoint):
+    for fstab_entry in fstab_entries:
+        if fstab_entry.fs_file == mountpoint:
+            return fstab_entry
+    return None
+
+
+def check_noexec_on_var(storage_info):
+    """Check for /var or /var/lib being mounted with noexec mount option."""
+
+    # Order of checking is important as mount options on /var/lib override those on /var
+    mountpoints_to_check = ('/var/lib/leapp', '/var/lib', '/var')
+    for mountpoint in mountpoints_to_check:
+        fstab_entry = find_fstab_entry_with_mountpoint(storage_info.fstab, mountpoint)
+        if fstab_entry and 'noexec' in fstab_entry.fs_mntops.split(','):
+            inhibit_upgrade_due_var_with_noexec(fstab_entry.fs_file, found_in_fstab=True)
+            return  # Do not check further as present mounts would likely reflect fstab
+
+    # Make sure present mounts doesn't contain noexec as wells - likely user mounted a partition
+    # and without adding it to fstab
+    for mountpoint in mountpoints_to_check:
+        mount_entry = find_mount_entry_with_mountpoint(storage_info.mount, mountpoint)
+        if mount_entry and 'noexec' in mount_entry.options.split(','):
+            inhibit_upgrade_due_var_with_noexec(mount_entry.mount, found_in_fstab=True)
```

Comment:
it should be `False` here
```suggestion
            inhibit_upgrade_due_var_with_noexec(mount_entry.mount, found_in_fstab=False)
```

Reply:
Thank you, fixed.